snowflake/cmd/trafficDrop.go

```diff
@@ -0,0 +1,294 @@
+	if endTs != "" {
+		endTime = endTs
+	} else if end != "" {
+		var err error
+		endTime, err = timestamps.ParseTimestamp(end, now)
+		if err != nil {
+			return "", err
+		}
+	}
+	if endTime != "" {
+		fmt.Fprintf(&queryBuilder, `AND
+  flowEndSeconds < '%s'
+`, endTime)
+	}
+
+	if clusterUUID != "" {
+		_, err := uuid.Parse(clusterUUID)
+		if err != nil {
+			return "", err
+		}
+		fmt.Fprintf(&queryBuilder, `AND
+  clusterUUID = '%s'
+`, clusterUUID)
+	} else {
+		logger.Info("No clusterUUID input, all flows will be considered during abnormal traffic drop detection.")
+	}
+
+	queryBuilder.WriteString(`GROUP BY
+	sourceIP,
+	sourcePodName,
+	sourcePodNamespace,
+	destinationIP,
+	destinationPodName,
+	destinationPodNamespace,
+	flowStartDate,
+	ingressNetworkPolicyRuleAction,
+	egressNetworkPolicyRuleAction
+), processed_flows AS (
+SELECT
+CASE WHEN f.ingressNetworkPolicyRuleAction = 2 OR f.ingressNetworkPolicyRuleAction = 3
+	THEN
+	(CASE WHEN f.destinationPodName IS NOT NULL
+		THEN CONCAT(f.destinationPodNamespace, '#', f.destinationPodName)
+		ELSE f.destinationIP
+	END)
+ELSE
+	(CASE WHEN f.sourcePodName IS NOT NULL
+		THEN CONCAT(f.sourcePodNamespace, '#', f.sourcePodName)
+		ELSE f.sourceIP
+	END)
+END as endpoint,
+CASE WHEN f.ingressNetworkPolicyRuleAction = 2 OR f.ingressNetworkPolicyRuleAction = 3
+	THEN
+	'ingress'
+ELSE
+	'egress'
+END as direction,
+f.flowStartDate as date,
+f.flowNumber as dropNumber
+FROM filtered_flows AS f
+`)
+
+	// Aggregate traffic drop number for each endpoint and direction
+	queryBuilder.WriteString(`), aggregated_flows as (
+SELECT 
+	pf.endpoint,
+	pf.direction,
+	pf.date, 
+	SUM(pf.dropNumber) as dropNumber
+FROM processed_flows AS pf
+GROUP BY
+	pf.endpoint,
+	pf.direction,
+	pf.date
+`)
+
+	// Choose the endpoint + direction as the partition field for the trafficDrop UDTF
+	// because traffic drop detection are worked on ingress and egress direction for each endpoint.
+	functionName := udfs.GetFunctionName(trafficDropFunctionName, functionVersion)
+	fmt.Fprintf(&queryBuilder, `)
+SELECT 
+	r.jobType, 
+	r.detectionID, 
+	r.timeCreated, 
+	r.endpoint,
+	r.direction,
+	r.avgDrop,
+	r.stdevDrop,
+	r.anomalyDropDate,
+	r.anomalyDropNumber FROM aggregated_flows AS af,
+TABLE(%s(
+	'%s',
+	'%s',
+	af.endpoint,
+	af.direction,
+	af.date, 
+	af.dropNumber
+) over (partition by af.endpoint, af.direction)) as r
+`, functionName, jobType, detectionID)
+
+	return queryBuilder.String(), nil
+}
+
+// trafficDropCmd represents the traffic-drop command
+var trafficDropCmd = &cobra.Command{
+	Use:   "traffic-drop",
+	Short: "Run the abnormal traffic drop detection UDF in Snowflake",
+	Long: `This command runs the abnormal traffic drop detection UDF in
+Snowflake. You need to bring your own Snowflake account and run the onboard
+command first.
+
+Run abnormal traffic drop detection with default configuration on database ANTREA_C9JR8KUKUIV4R72S:
+"theia-sf traffic-drop --database-name ANTREA_C9JR8KUKUIV4R72S"
```

Comment:
do you think `traffic-drop-detection` would be too long for the command name?
what about `drop-detection`?

I feel like overall, "drop detection" may be a better name for this feature (compared to "traffic drop), hence for the command and for the UDTF as well

Reply:
Sounds good to me, changed it to `drop-detection`.